Source: Alpine Linux
Upstream: Unknown
Reason: Fixes compilation with musl libc
---

diff --git a/parser/missingdefs.h b/parser/missingdefs.h
new file mode 100644
index 0000000..8097aef
--- /dev/null
+++ b/parser/missingdefs.h
@@ -0,0 +1,8 @@
+#ifndef PARSER_MISSINGDEFS_H
+#define PARSER_MISSINGDEFS_H
+
+typedef int (*__compar_fn_t) (const void *, const void *);
+typedef __compar_fn_t comparison_fn_t;
+typedef void (*__free_fn_t) (void *__nodep);
+
+#endif
diff --git a/parser/parser_alias.c b/parser/parser_alias.c
index f5b6da4..d57f580 100644
--- a/parser/parser_alias.c
+++ b/parser/parser_alias.c
@@ -25,6 +25,10 @@
 #include "parser.h"
 #include "profile.h"
 
+#ifndef __GLIBC__
+#include "missingdefs.h"
+#endif
+
 struct alias_rule {
 	char *from;
 	char *to;
diff --git a/parser/parser_symtab.c b/parser/parser_symtab.c
index 3e667d8..e109f4d 100644
--- a/parser/parser_symtab.c
+++ b/parser/parser_symtab.c
@@ -25,6 +25,10 @@
 #include "immunix.h"
 #include "parser.h"
 
+#ifndef __GLIBC__
+#include "missingdefs.h"
+#endif
+
 enum var_type {
 	sd_boolean,
 	sd_set,
